browser/ui/views/tabs/brave_alert_indicator.cc

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/tabs/brave_alert_indicator.h"
+
+#include <string>
+
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "chrome/browser/ui/views/tabs/browser_tab_strip_controller.h"
+#include "chrome/browser/ui/views/tabs/tab_controller.h"
+#include "chrome/browser/ui/views/tabs/tab_strip.h"
+#include "chrome/browser/ui/views/tabs/tab_strip_controller.h"
+#include "chrome/browser/ui/views/tabs/tab_style_views.h"
+#include "content/public/browser/web_contents.h"
+
+namespace {
+
+bool IsAudioState(TabAlertState state) {
+  return (state == TabAlertState::AUDIO_PLAYING ||
+          state == TabAlertState::AUDIO_MUTING);
+}
+
+}  // namespace
+
+SkColor BraveAlertIndicator::GetBackgroundColor() const {
+  TabStyle::TabColors colors = parent_tab_->tab_style()->CalculateColors();
+  return mouse_pressed_ ? colors.button_background_pressed_color
+                        : colors.button_background_hovered_color;
+}
+
+bool BraveAlertIndicator::OnMousePressed(const ui::MouseEvent& event) {
+  mouse_pressed_ = true;
+  SchedulePaint();
+
+  if (!IsAudioState(alert_state_))
+    return AlertIndicator::OnMousePressed(event);
+
+  return true;
+}
+
+void BraveAlertIndicator::OnMouseReleased(const ui::MouseEvent& event) {
+  mouse_pressed_ = false;
+  SchedulePaint();
+
+  if (!IsAudioState(alert_state_) || !IsMouseHovered())
+    return AlertIndicator::OnMouseReleased(event);
+
+  auto* tab_strip = static_cast<TabStrip*>(parent_tab_->controller());
+  const int tab_index = tab_strip->GetModelIndexOfTab(parent_tab_);
+  auto* tab_strip_model = static_cast<BrowserTabStripController*>(
+      tab_strip->controller())->model();
+  auto* web_contents = tab_strip_model->GetWebContentsAt(tab_index);
+
+  if (!chrome::CanToggleAudioMute(web_contents))
+    return AlertIndicator::OnMouseReleased(event);
+
+  chrome::SetTabAudioMuted(web_contents,
+                           !web_contents->IsAudioMuted(),
+                           TabMutedReason::CONTEXT_MENU,
+                           std::string());
+}
+
+void BraveAlertIndicator::OnPaintBackground(gfx::Canvas* canvas) {
```

Comment:
why do we have changes to painting?

Reply:
because upstream icon doesn't have any hover/click effect on speaker icon.

Reply:
Seems like we should be calling `SetBackground` instead of overriding `OnPaintBackground`? Or maybe it makes more sense to do `SetBorder`?

Reply:
I followed close button style. but maybe we can do same thing with `Background`. Do you think using `Background` is better?

Reply:
In this case, I think there is no difference between overriding `OnPaintBackground` and `SetBackground`.
If we want to show hover/press effect with border, we could use `Border`.

Reply:
I think `SetBackground`/`SetBorder` seem safer

Reply:
@bridiver Changed to `Background`.